### PR TITLE
Add Support for mTLS Authentication 

### PR DESCRIPTION
**Pull request by oliverisaac**
### Motivation

Add support for mTLS Auth as that is what we are using for authentication with pulsar.

### Modifications

- Added the `tls_cert_file_path` and `tls_key_file_path` option. 

- A tiny bit of cleanup on the readme

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is already covered by the existing provider tests

### Documentation

- [ ] `doc-required`  
- [ ] `no-need-doc`  
- [x] `doc`

### CODE REVIEW

**Comment by LanceSandino** [comment]
Thanks, my company really needs this! 👍🏽

File: pulsar/provider.go [hunk 1]
```diff
@@ -98,12 +102,24 @@ func Provider() *schema.Provider {
 				DefaultFunc: schema.MultiEnvDefaultFunc(
 					[]string{"PULSAR_TLS_TRUST_CERTS_FILE_PATH", "TLS_TRUST_CERTS_FILE_PATH"}, ""),
 			},
+			"tls_cert_file_path": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: descriptions["tls_cert_file_path"],
+				DefaultFunc: schema.MultiEnvDefaultFunc([]string{"PULSAR_TLS_CERT_FILE", "PULSAR_TLS_CERT_FILE_PATH"}, ""),
```
Comment (nodece) [inline pulsar/provider.go:109]:
```suggestion
				DefaultFunc: schema.MultiEnvDefaultFunc([]string{"PULSAR_TLS_CERT_FILE_PATH"}, ""),
```

File: pulsar/provider.go [hunk 2]
```diff
@@ -98,12 +102,24 @@ func Provider() *schema.Provider {
...
+			},
+			"tls_key_file_path": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: descriptions["tls_key_file_path"],
+				DefaultFunc: schema.MultiEnvDefaultFunc([]string{"PULSAR_TLS_KEY_FILE", "PULSAR_TLS_KEY_FILE_PATH"}, ""),
```
Comment (nodece) [inline pulsar/provider.go:115]:
```suggestion
				DefaultFunc: schema.MultiEnvDefaultFunc([]string{"PULSAR_TLS_KEY_FILE_PATH"}, ""),
```

**Review by nodece — CHANGES_REQUESTED** [review]
Good work!

**Comment by nodece** [comment]
Could you remove `.DS_Store` file?

**Comment by oliverisaac** [comment]
Thanks @nodece! Got those changes dropped in.

**Review by maxsxu — APPROVED** [review]
@oliverisaac LGTM. Thanks for your contributions!

**Review by maxsxu — APPROVED** [review]

**Pull request merged**